client/my-sites/email/email-management/home/email-plan-warnings.tsx

```diff
@@ -21,17 +28,45 @@ type EmailPlanWarningsProps = {
 
 const EmailPlanWarnings = ( { domain, emailAccount }: EmailPlanWarningsProps ) => {
 	const translate = useTranslate();
-	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+	const selectedSite = useSelector( getSelectedSite );
+	const selectedSiteSlug = selectedSite?.slug ?? '';
 
 	const warning = emailAccount?.warnings?.[ 0 ];
 
+	const cannotAddEmailWarningReason = getCurrentUserCannotAddEmailReason( domain );
+	const cannotAddEmailWarningMessage = cannotAddEmailWarningReason?.message ?? '';
+	const cannotAddEmailWarningCode = cannotAddEmailWarningReason?.code ?? null;
+
+	const WarningMessageForCode = useMemo( () => {
```

Comment:
I think we should be able to do this without `useMemo`. Off the top of my head, my preferred solution would probably be to just make `WarningMessageForCode` a regular component and define it outside of the `EmailPlanWarnings` scope (i.e. quite similar to what we had in f774c23bfab20f0ffdbab108db19a1c0b7198b8c, except with a different scope).

@wongasy AFAICT, it's not an issue if `useDomainOwnerUserName` is called repeatedly. `react-query` has internal state management that will only make the required network requests if they aren't already in progress.